content/v2.2/cli-commands/db.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Routes
+order: 110
+---
+
+## hanami db
+
+Hanami supports multiple database-related CLI commands, to help manage databases and their schemas for your app. You can list all of them, by adding the `--help` option.
+
+```shell
+$ bundle exec hanami db --help
+
+Commands:
+  hanami db create                                  # Create databases
+  hanami db drop                                    # Delete databases
+  hanami db migrate                                 # Migrates database
+  hanami db prepare                                 # Prepare databases
+  hanami db seed                                    # Load seed data
+  hanami db structure [SUBCOMMAND]
+  hanami db version                                 # Print schema version
+```
+
+### hanami db create
+Allows to create the database for the current environment.
+
+```shell
+$ bundle exec hanami db create
+
+# => database db/bookshelf.sqlite created
+```
+
+The DB name is fetched from the `DATABASE_URL` environment variable.
+
+```yaml
+# .env
+DATABASE_URL=sqlite://db/guides_app.sqlite
+```
+
+The command will accept options listed below.
+
+```shell
+  --app                             # Use app database, default: false
+  --slice=VALUE                     # Use database for slice
+  --gateway=VALUE                   # Use database for gateway
+  --help, -h                        # Print this help
+```
+
+**Create slice database**
+
+You can have separate databases for your slices without any additional configuration. The only thing needed is to define the `DATABASE_URL` environment variable for your slice. You can do it, by prepending the `DATABASE_URL` with your slice name, separating it by `__` (**double underscore**). Here is a quick example:
+
+```shell
+# quickly generate the new slice if you don't have one.
+bundle exec hanami g slice Admin
+```
+
+```yaml
+# .env
+DATABASE_URL=sqlite://db/guides_app.sqlite
+
+ADMIN__DATABASE_URL=sqlite://db/admin_guides_app.sqlite
+```
+
+Now, when you'll run the command with the `--slice` option, separate db will be created for your slice.
+
+```shell
+$ bundle exec hanami db create
+
+# => database db/admin_bookshelf.sqlite created
+```
+
+TODO: `--app` option. How this is supposed to work?
```

Comment:
@timriley It is unclear to me how to use this?

When I specify the `--slice` and `-app`, it seems to be ignored.

```shell
bundle exec hanami db create --slice=Main
=> database db/main_guides_app.sqlite created

bundle exec hanami db create --slice=Main --app
=> database db/main_guides_app.sqlite created
```

Could you please help me understand the context of this `--app` option?

Reply:
`--app` would be the top-level `Hanami::App` rather than a slice. So `--app` and `--slice` are mutually-exclusive.

Reply:
This is what I thought, but it's unclear how to use it. Not passing an option generates the db for app,, doesn't it? Or do I miss the env variable for app?

Reply:
Ah, perhaps Tim intends these to be _additive_, so you can process multiple db configs at once

Reply:
@swilgosz my intention is that you run these commands:

- _without_ `--app` or `--slice` options (so they work on _all_ databases)
- or with only a _single_ `--app` or `--slice` option (so you can work on a particular subset of databases)

I didn't consider both being provided. I'm pretty sure the implementation would just end up considering one of the args in that case.

(We should probably update our option handling to make that clear)